### PR TITLE
Support a standard go path by using "go env GOPATH"

### PR DESCRIPTION
**Pull request by djeebus**
Also cache it for faster repeat lookups

### CODE REVIEW

File: utils.go [hunk 1]
```diff
@@ -397,10 +424,17 @@ func doesPluginSourceExist(gitURL string) (exists bool) {
 	return info.IsDir()
 }
 
-func getGoPath() (gopath string) {
-	if gopath = os.Getenv("GOPATH"); len(gopath) > 0 {
-		return
-	}
+var goPath = ""
```
Comment (dhalman) [inline utils.go]: @djeebus we usually do globals at the top (if we can't avoid them)

Also, we try to do named return values 
`getGoPath() (goPath string, err error)`

Then set those values upon return

```
if len(cachedGoPath) == 0 {
   // Parse goPath
   ...
   if err != nil {
      return
   }
}

goPath = cachedGoPath
return
```

File: utils.go [hunk 2]
```diff
@@ -34,7 +33,10 @@ func ParseKey(key string) (newKey, alias string) {
 
 func gitFetchTags(gitURL string) (err error) {
 	gitfetch := exec.Command("git", "fetch", "--tags", "--force")
-	gitfetch.Dir = getGitDir(gitURL)
+	gitfetch.Dir, err = getGitDir(gitURL)
```
Comment (itsmontoya) [inline utils.go]:
```go
if gitfetch.Dir, err = getGitDir(gitURL); err != nil {
   err = fmt.Errorf("error getting git directory for URL \"%s\": %v", gitURL, err)
   return
}
```

Comment (itsmontoya) [reply utils.go]: Also, add newline before `gitfetch.Stdin = os.Stdin`

File: utils.go [hunk 3]
```diff
@@ -52,7 +54,10 @@ func gitFetchTags(gitURL string) (err error) {
 
 func gitCheckout(gitURL, branch string) (resp string, err error) {
 	gitcheckout := exec.Command("git", "checkout", branch)
-	gitcheckout.Dir = getGitDir(gitURL)
+	gitcheckout.Dir, err = getGitDir(gitURL)
```
Comment (itsmontoya) [inline utils.go]:
```go
if gitcheckout.Dir, err = getGitDir(gitURL); err != nil {
   err = fmt.Errorf("error getting git directory for URL \"%s\": %v", gitURL, err)
   return
}
```

File: utils.go [hunk 4]
```diff
@@ -85,7 +90,10 @@ func gitCheckout(gitURL, branch string) (resp string, err error) {
 
 func gitPull(gitURL string) (resp string, err error) {
 	gitpull := exec.Command("git", "pull", "origin")
-	gitpull.Dir = getGitDir(gitURL)
+	gitpull.Dir, err = getGitDir(gitURL)
```
Comment (itsmontoya) [inline utils.go]:
```go
if gitpull.Dir, err = getGitDir(gitURL); err != nil {
   err = fmt.Errorf("error getting git directory for URL \"%s\": %v", gitURL, err)
   return
}
```

File: utils.go [hunk 5]
```diff
@@ -116,7 +124,10 @@ func updatePluginDependencies(gitURL string) (err error) {
 	update := exec.Command("go", args...)
 	update.Stdin = os.Stdin
 	update.Stdout = os.Stdout
-	update.Dir = getGoDir(gitURL)
+	update.Dir, err = getGoDir(gitURL)
```
Comment (itsmontoya) [inline utils.go]:
```go
if update.Dir, err = getGitDir(gitURL); err != nil {
   err = fmt.Errorf("error getting git directory for URL \"%s\": %v", gitURL, err)
   return
}
```

**Review by itsmontoya — CHANGES_REQUESTED** [review]
Left some notes!

**Comment by dhalman** [comment]
@djeebus @itsmontoya are we still planning on getting this updated and merged?

**Comment by djeebus** [comment]
Just noticed the new comments. I'll get them sorted out soon.

**Comment by djeebus** [comment]
Ok, comments resolved.

File: utils.go [hunk 6]
```diff
@@ -423,19 +425,22 @@ func doesPluginSourceExist(gitURL string) (exists bool) {
...
 		output, err := goEnvCmd.CombinedOutput()
 		if err != nil {
-			return "", err
+			err = fmt.Errorf("failed to read GOPATH: %v", err)
+			return
 		}
 
-		goPath = strings.TrimSpace(string(output))
+		cachedGoPath = strings.TrimSpace(string(output))
+		goPath = cachedGoPath
```
Comment (dhalman) [inline utils.go:440]: I'd remove line 429 and move this to before the final return statement (outside the conditional)

**Pull request closed without merging**